lib/auth/clt.go

```diff
@@ -2716,494 +2037,26 @@ func (c *Client) ValidateTrustedCluster(validateRequest *ValidateTrustedClusterR
 	return validateResponse, nil
 }
 
-// DeleteTrustedCluster deletes a trusted cluster by name.
-func (c *Client) DeleteTrustedCluster(ctx context.Context, name string) error {
-	_, err := c.Delete(c.Endpoint("trustedclusters", name))
-	return trace.Wrap(err)
-}
-
-func (c *Client) GetAccessRequests(ctx context.Context, filter services.AccessRequestFilter) ([]services.AccessRequest, error) {
-	clt, err := c.grpc()
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
-	rsp, err := clt.GetAccessRequests(ctx, &filter)
-	if err != nil {
-		return nil, trail.FromGRPC(err)
-	}
-	reqs := make([]services.AccessRequest, 0, len(rsp.AccessRequests))
-	for _, req := range rsp.AccessRequests {
-		reqs = append(reqs, req)
-	}
-	return reqs, nil
-}
-
-func (c *Client) CreateAccessRequest(ctx context.Context, req services.AccessRequest) error {
-	r, ok := req.(*services.AccessRequestV3)
-	if !ok {
-		return trace.BadParameter("unexpected access request type %T", req)
-	}
-	clt, err := c.grpc()
-	if err != nil {
-		return trace.Wrap(err)
-	}
-	_, err = clt.CreateAccessRequest(ctx, r)
-	if err != nil {
-		return trail.FromGRPC(err)
-	}
-	return nil
-}
-
-func (c *Client) RotateResetPasswordTokenSecrets(ctx context.Context, tokenID string) (services.ResetPasswordTokenSecrets, error) {
-	clt, err := c.grpc()
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
-
-	secrets, err := clt.RotateResetPasswordTokenSecrets(ctx, &proto.RotateResetPasswordTokenSecretsRequest{
-		TokenID: tokenID,
-	})
-	if err != nil {
-		return nil, trail.FromGRPC(err)
-	}
-
-	return secrets, nil
-}
-
-func (c *Client) GetResetPasswordToken(ctx context.Context, tokenID string) (services.ResetPasswordToken, error) {
-	clt, err := c.grpc()
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
-
-	token, err := clt.GetResetPasswordToken(ctx, &proto.GetResetPasswordTokenRequest{
-		TokenID: tokenID,
-	})
-	if err != nil {
-		return nil, trail.FromGRPC(err)
-	}
-
-	return token, nil
-}
-
 // CreateResetPasswordToken creates reset password token
 func (c *Client) CreateResetPasswordToken(ctx context.Context, req CreateResetPasswordTokenRequest) (services.ResetPasswordToken, error) {
-	clt, err := c.grpc()
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
-
-	token, err := clt.CreateResetPasswordToken(ctx, &proto.CreateResetPasswordTokenRequest{
+	return c.APIClient.CreateResetPasswordToken(ctx, authProto.CreateResetPasswordTokenRequest{
```

Comment:
Can you use a type alias here?

Reply:
https://github.com/gravitational/teleport/pull/5091